dev/pyproject.py

```diff
@@ -28,6 +34,19 @@ def build(skinny: bool) -> None:
         r'^VERSION = "([a-z0-9\.]+)"$', Path("mlflow", "version.py").read_text(), re.MULTILINE
     ).group(1)
     python_version = Path("requirements", "python-version.txt").read_text().strip()
+    versions_yaml = read_package_versions_yml()
+    langchain_requirements = [
+        "langchain>={},<{}".format(
+            max(
+                versions_yaml["langchain"]["autologging"]["minimum"],
+                versions_yaml["langchain"]["models"]["minimum"],
```

Comment:
```suggestion
                Version(versions_yaml["langchain"]["autologging"]["minimum"]),
                Version(versions_yaml["langchain"]["models"]["minimum"]),
```

can we use compare the versions as `packaing.version.Version`?